odoo/addons/base/models/ir_mail_server.py

```diff
@@ -216,7 +224,17 @@ def _get_test_email_addresses(self):
                               'sending an email message via this outgoing server'))
         return email_from, 'noreply@odoo.com'
 
-    def test_smtp_connection(self):
+    def test_smtp_connection(self, autodetect_max_email_size=False):
+        """Test the connection and if autodetect_max_email_size, set auto-detected max email size.
+
+        The method raises UserError
```

Comment:
There is a standard way of indicating exceptions in docstrings using ':raises', like
```
:raises:
    UserError: Explanation here.
```